microprofile/opentracing-smallrye/src/main/java/org/wildfly/microprofile/opentracing/smallrye/TracingCDIExtension.java

```diff
@@ -57,6 +71,17 @@ public void skipTracerBeans(@Observes ProcessAnnotatedType<? extends Tracer> pro
      * @param bs
      */
     public void beforeShutdown(@Observes final BeforeShutdown bs) {
-        TRACERS.remove(Thread.currentThread().getContextClassLoader());
+        ClassLoader loader;
```

Comment:
This entire block can be removed and instead just call WildFlySecurityManager.getCurrentContextClassLoaderPrivileged()

Additionally with this current approach two new classes needed to be defined and loaded to achieve a common task.